src/ruby/yast/y2base.rb

```diff
@@ -0,0 +1,131 @@
+      ret[:client_options] = parse_client_options(args)
+      ret[:server_name] = args.shift or raise "Missing server name."
+      ret[:server_options] = args
+
+      ret
+    end
+
+    def self.help
+      "Usage: y2base [GenericOpts] Client [ClientOpts] Server " \
+      "[Specific ServerOpts]\n" \
+      "\n" \
+      "GenericOptions are:\n" \
+      "    -h --help         : Sprint this help\n" \
+      "\n" \
+      "ClientOptions are:\n" \
+      "    -a --arg          : add argument for client. Can be used multiple times.\n"
+      "\n" \
+      "Specific ServerOptions are any options passed on unevaluated.\n" \
+      "\n" \
+      "Examples:\n" \
+      "y2base installation qt\n" \
+      "    Start binary y2base with intallation.ycp as client and qt as server\n" \
+      "y2base installation -a initial qt\n" \
+      "    Provide parameter initial for client installation\n" \
+      "y2base installation qt -geometry 800x600\n" \
+      "    Provide geometry information as specific server options\n"
+    end
+
+    # so how works signals now in ruby?
+    # it logs what we know about signal and then continue with standard ruby
+    # handler, so raising {SignalException} that can be processed
+    def self.setup_signals
+      Signal.trap("PIPE", "IGNORE")
+
+      # SEGV, ILL and FPE is reserved, so cannot be set
+      ["HUP", "INT", "QUIT", "ABRT", "TERM"].each do |name|
+        Signal.trap(name) { signal_handler(name) }
+      end
+    end
+
+    private_class_method def self.signal_handler(name)
+      Signal.trap(name, "IGNORE")
+
+      $stderr.puts "YaST got signal #{name}."
+
+      signal_log_open do |f|
+        f.puts "=== #{Time.now} ==="
+        f.puts "YaST got signal #{name}."
+        # TODO: print stored debug logs
+        f.puts "Backtrace (only ruby one):"
+        caller.each { |l| f.puts(l) }
+      end
+
+      system("/usr/lib/YaST2/bin/signal-postmortem")
+
+      Signal.trap(name, "DEFAULT")
+      Process.kill(name, Process.pid)
+    end
+
+    LOG_LOCATIONS = ["/var/log/YaST2/signal", "y2signal.log"]
+    private_class_method def self.signal_log_open(&block)
+      index = 0
+      begin
+        path = LOG_LOCATIONS[index]
+        return unless path
+        File.open(path, "a") { |f| block.call(f) }
+      rescue IOError, SystemCallError
+        index +=1
+        retry
+      end
+    end
+
+    private_class_method def self.parse_generic_options(args)
+      res = {}
+      loop do
+        return res unless option?(args.first)
+
+
+        arg = args.shift
+        case arg
+        when "-h", "--help"
+          res[:help] = true
+        else
+          raise "Unknown option #{args.first}"
+        end
+      end
+    end
+
+    private_class_method def self.parse_client_options(args)
+      res = {}
+      res[:params] = []
+      loop do
+        return res unless option?(args.first)
+
+        arg = args.shift
+        case arg
+        when "-a", "--arg"
```

Comment:
changelog:
- changed the syntax for client arguments
y2base foo -S bar qt → y2base foo --arg bar qt